File: problem3/problem3.py
def is_prime(n):
    i = 3
    if n % 2 == 0:
        return False
    while i*i < n:
        if n % i == 0:
            return False
        i += 2
    return True
            
def problem3(n):
    largest_factor = 0
    i = 3
    while i*i < n:
        if n % i == 0 and is_prime(i):
            largest_factor = i
        i += 2
    return largest_factor

if __name__ == '__main__':
    print(problem3(600851475143))

File: problem3/problem3.cpp
#include <iostream>

bool is_prime(long long n)
{
	unsigned int i = 3;
	if (n % 2 == 0)
		return false;
	while (i*i < n)
	{
		if (n % i == 0)
			return false;
		i += 2;
	}
	return true;
}

unsigned int problem3(long long n)
{
	unsigned int largest_factor = 0;
	long long i = 3;
	while (i*i < n)
	{
		if (n % i == 0 && is_prime(i))
			largest_factor = i;
		i += 2;
	}
	return largest_factor;
}

int main()
{
	std::cout << problem3(600851475143) << std::endl;
	return 0;
}